Simulation/LinearBatteryModel_Generator.m

```diff
@@ -0,0 +1,25 @@
+function [Gen_LinearBatteryModel] = LinearBatteryModel_Generator(MinVoltage, MaxVoltage, MaxCapacity)
+  % This function generates a function delegate using the input parameters for a Linear Battery
+  % Model meaning that voltage is directly proportional to the remaining capacity.
+  % The model calculates the voltage by linearly interpolating it between Min and Max voltage based
+  % on the provided Capacity.
+  %
+  % Parameters:
+  %     MinVoltage  - Minimum voltage of the battery at the lowest (safe) capacity (V)
+  %     MaxVoltage  - Maximum voltage of the battery at full capacity (V)
+  %     MaxCapacity - Maximum capacity at the battery (Ah)
+  %
+  % Example:
+  %     PlaneBattery = BatteryModel_Generator(MaxVoltage, MinVoltage, Capacity);
+  %     ThrustModel = ElectricPropellerModel_Generator(..., PlaneBattery);
+  %     ServoModel  = ServoModel_Generator(..., PlaneBattery);
+
+
+  Gen_LinearBatteryModel = @LinearBatteryModel;
+
+  function Voltage = LinearBatteryModel(Capacity)
+    SV_Index = StateVector_Index;
+
+    Voltage = interp1([0, MaxCapacity], [MinVoltage, MaxVoltage], Capacity);

```

Comment:
Just confirming, this will not extrapolate outside the bounds of 0 and MaxCapacity and will actually output a value (the min and max voltages respectively) if outside the bounds?

Reply:
Looks like `interp1` returns `NA` if the `Capacity` is outside of the specified range. We could add validation to make sure this doesn't happen to this model, but validating the whole state vector in one centralized place before passing it to different models would be a much better idea.

Reply:
The validation should be at the model level, since that puts the burden on the user to make sure their model is accurate and reflects the desired behaviour.

I have no clue how we could create a validation system that is either generic or something someone else must make, personally, but if you have some idea that would be easy to do and not much added workload for the users let me know because I would like to try to take that load off the user (these sort of "fringe" cases are usually a source of "gotchas" in modeling)